planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/src/scene_module/crosswalk/scene_crosswalk.cpp

```diff
@@ -197,57 +182,172 @@ bool CrosswalkModule::checkStopArea(
+          *planner_data_, output, debug_data_, first_stop_path_point_index_))
+      {
+        return false;
+      }
+    } else {
+      if (!insertTargetVelocityPoint(
+          input, slowdown_polygon, planner_param_.stop_line_distance + planner_param_.slow_margin,
+          planner_param_.slow_velocity, *planner_data_, output, debug_data_,
+          first_stop_path_point_index_))
+      {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+bool CrosswalkModule::createVehiclePathPolygonInCrosswalk(
+  const autoware_planning_msgs::msg::PathWithLaneId & input,
+  const boost::geometry::model::polygon<boost::geometry::model::d2::point_xy<double>>
+  & crosswalk_polygon,
+  const float extended_width, Polygon & path_polygon)
+{
+  std::vector<Point> path_collision_points;
+  for (size_t i = 0; i < input.points.size() - 1; ++i) {
+    const auto p0 = input.points.at(i).point.pose.position;
+    const auto p1 = input.points.at(i + 1).point.pose.position;
+    const Line line{{p0.x, p0.y}, {p1.x, p1.y}};
+    std::vector<Point> line_collision_points;
+    bg::intersection(crosswalk_polygon, line, line_collision_points);
+    if (line_collision_points.empty()) {continue;}
+    for (size_t j = 0; j < line_collision_points.size(); ++j) {
+      path_collision_points.push_back(line_collision_points.at(j));
+    }
+  }
+  if (path_collision_points.size() != 2) {
+    RCLCPP_ERROR_THROTTLE(
+      logger_, *clock_, 5.0,
+      "There must be two points of conflict between the crosswalk polygon and the path. points is "
+      "%d",
+      (int)path_collision_points.size());
+    return false;
+  }
+
+  Polygon candidate_path_polygon;
   {
+    const double width = planner_data_->vehicle_info_.vehicle_width_m_;
+    const double d = (width / 2.0) + extended_width;
+    const auto cp0 = path_collision_points.at(0);
+    const auto cp1 = path_collision_points.at(1);
+    const double yaw = std::atan2(cp1.y() - cp0.y(), cp1.x() - cp0.x()) + M_PI_2;
+    const double dcosyaw = d * std::cos(yaw);
+    const double dsinyaw = d * std::sin(yaw);
+    candidate_path_polygon.outer().push_back(bg::make<Point>(cp0.x() + dcosyaw, cp0.y() + dsinyaw));
+    candidate_path_polygon.outer().push_back(bg::make<Point>(cp1.x() + dcosyaw, cp1.y() + dsinyaw));
+    candidate_path_polygon.outer().push_back(bg::make<Point>(cp1.x() - dcosyaw, cp1.y() - dsinyaw));
+    candidate_path_polygon.outer().push_back(bg::make<Point>(cp0.x() - dcosyaw, cp0.y() - dsinyaw));
+    candidate_path_polygon.outer().push_back(candidate_path_polygon.outer().front());
+  }
+  candidate_path_polygon = isClockWise(candidate_path_polygon) ?
+    candidate_path_polygon :
+    inverseClockWise(candidate_path_polygon);
+
+  std::vector<Polygon> path_polygons;
+  bg::intersection(crosswalk_polygon, candidate_path_polygon, path_polygons);
+
+  if (path_polygons.size() != 1) {
+    RCLCPP_ERROR_THROTTLE(
+      logger_, *clock_, 5.0, "Number of polygon is %d. Must be 1",
```

Comment:
The unit for RCLCPP_ERROR_THROTTLE is ms.
```suggestion
      logger_, *clock_, 5000, "Number of polygon is %d. Must be 1",
```

Reply:
fixed at 4c139da
Also I check other throttle too.